src/main/java/com/patchworkmc/annotation/AnnotationProcessor.java

```diff
@@ -12,18 +12,35 @@
 
 public class AnnotationProcessor extends ClassVisitor {
 	private Consumer<String> consumer;
-
-	public AnnotationProcessor(ClassVisitor parent, Consumer<String> consumer) {
+	private AnnotationStorage annotationStorage;
+	private String className;
+
+	public AnnotationProcessor(
+			ClassVisitor parent,
+			Consumer<String> consumer,
+			AnnotationStorage annotationStorage
+	) {
 		super(Opcodes.ASM7, parent);
 
 		this.consumer = consumer;
+		this.annotationStorage = annotationStorage;
 	}
 
 	private static boolean isKotlinMetadata(String descriptor) {
 		// TODO: This is specific to one mod
 		return descriptor.startsWith("Lcom/greenapple/glacia/embedded/kotlin/");
 	}
 
+	private static boolean isForgeAnnotation(String descriptor) {
+		return descriptor.contains("forge");
```

Comment:
This could result in false positives, a more selective match would probably be good.
```suggestion
		return descriptor.startsWith("Lnet/minecraftforge/");
```